packages/headless/src/modules/cart/services/index.ts

```diff
@@ -0,0 +1,165 @@
+import { HttpStatus, Injectable } from '@nestjs/common';
+import { Helper } from 'src/helper/helper.interface';
+import { CartRepository } from '../repositories';
+import { AddToCartResponse, getCartResponse, ErrorMessage, getCartErrorMessage, deleteCartErrorMessage, updateCartItemErrorMessage, deleteCartItemErrorMessage, deleteAllCartItemsErrorMessage, deleteCartResponse, updateCartItemResponse, deleteCartItemResponse, deleteAllCartItemsResponse } from 'models';
+import { Item, UpdateItem } from 'src/entity/cart';
+
+@Injectable()
+export class CartService {
+  constructor(private cartRepo: CartRepository, private helper: Helper) { }
+
+  async addToCart(
+    item: Item,
+    userId: string,
+  ): Promise<AddToCartResponse> {
+    const existCart = await this.cartRepo.isCartExist(userId);
+    if (!existCart) {
+      const createCart = await this.cartRepo.createCart({ userId, items: [item] });
+      if (!createCart) {
+        return this.helper.serviceResponse.errorResponse(
+          ErrorMessage.CANNOT_CREATE_CART,
+          null,
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
+      }
```

Comment:
avoid using nested if block. you could have written the first line like this
 const createCart = !existCart && await this.cartRepo.createCart({ userId, items: [item] });
then the second if block is okay.
nested blocks makes things ugly. so avoid as much as possible.